aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -81,6 +82,19 @@ pub fn allow_module_bundle_for_test() {
     MODULE_BUNDLE_DISALLOWED.store(false, Ordering::Relaxed);
 }
 
+/// Contains information about execution failure.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+struct ExecutionError {
+    abort_location: String,
+    // There are 3 error types, stored as strings:
+    // 1. VMError. Indicates an error from the VM, e.g. out of gas, invalid auth key, etc.
+    // 2. MoveAbort. Indicates an abort, e.g. assertion failure, from inside the executed Move code.
+    // 3. MoveExecutionFailure. Indicates an error from Move code where the VM could not continue. For example,
+    // arithmetic failures.
+    error_type: String,
```

Comment:
should this be an enum?

Reply:
Enum unfortunately doesn't serialize well to Move side. I want to have string on Move side for readability. I'd have to write a long custom serializer for this one case just to have it all :(